src/dbnode/persist/fs/retriever.go

```diff
@@ -342,7 +342,7 @@ func (r *blockRetriever) fetchBatch(
 		}
 
 		// We don't need to call onRetrieve.OnRetrieveBlock if the ID was not found.
-		callOnRetrieve := req.onRetrieve != nil && req.foundAndHasNoError()
+		callOnRetrieve := r.opts.CacheOnRetrieve() && req.onRetrieve != nil && req.foundAndHasNoError()
```

Comment:
nit: rename var to `cacheOnRetrieve`

Reply:
@arnikola and I synced offline. Per-namespace config is much safer imo, so will rework this PR with that change.

Reply:
Might want to also allow a global though if folks don't want to specify it per-namespace.

Reply:
A little bit reticent of giving people an easy knob that could accidentally hobble their query/alert perf, especially for unaggregated ns... what would you think about a "timeboxed" approach? I.e. you'd specify that you only want to cache blocks seen in the last e.g. 2 hours, and then that would also set cache expiry to the same value?

Reply:
Forgot to update this, but, for posterity, synced with @robskillington and @arnikola via Slack. Consensus was to add both a global and namespace-specific configuration option for disabling caching.